app/src/main/java/com/example/ubclaunchpad/launchnote/BaseActivity.kt

```diff
@@ -73,9 +75,37 @@ abstract class BaseActivity : AppCompatActivity() {
 
     abstract fun getContentViewId(): Int
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if(requestCode == TakePhotoActivity.TAKE_PHOTO_REQUEST_CODE || requestCode == GalleryActivity.GALLERY_ACTIVITY_REQ_CODE) {
+            when (resultCode) {
+                Activity.RESULT_OK -> {
+                    val picNote = data!!.getSerializableExtra(PIC_NOTE_KEY) as PicNote
+                    editPicNote(picNote)
+                }
+                Activity.RESULT_CANCELED -> {
+                    // todo vpineda what to do when the photo is not saved
```

Comment:
I'm thinking it should go back to PhotoBrowserActivity rather than exit the app entirely.